crates/lib3h_protocol/src/types.rs

```diff
@@ -0,0 +1,45 @@
+use holochain_persistence_api::hash::HashString;
+use std::fmt;
+
+//--------------------------------------------------------------------------------------------------
+// SpaceHash: newtype for HashString
+//--------------------------------------------------------------------------------------------------
+
+#[derive(PartialOrd, PartialEq, Eq, Ord, Clone, Debug, Serialize, Deserialize, Default, Hash)]
+pub struct SpaceHash(HashString);
+
```

Comment:
Consider also using [shrinkwraprs](https://lib.rs/crates/shrinkwraprs). This would be useful if we need to treat them as `HashString`s quite often and use whatever api `HashString` provides.